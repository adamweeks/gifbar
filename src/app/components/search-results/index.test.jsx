import React from 'react';
import renderer from 'react-test-renderer';

import SearchResults from '.';

describe(`SearchResults component`, () => {
  const image1 = {
    id: 1,
    displayUrl: `http://mycoolserver.info/kid-n-play.gif`,
    fullSizedImageFileSize: `5 Mb`,
    imageSizes: {
      smallSize: {
        height: 50
      }
    }
  };
  const image2 = {
    id: 2,
    displayUrl: `http://mycoolserver.info/kid-n-play.gif`,
    fullSizedImageFileSize: `5 Mb`,
    imageSizes: {
      smallSize: {
        height: 50
      }
    }
  };
  const image3 = {
    id: 3,
    displayUrl: `http://mycoolserver.info/kid-n-play.gif`,
    fullSizedImageFileSize: `5 Mb`,
    imageSizes: {
      smallSize: {
        height: 50
      }
    }
  };
  const copyUrl = jest.fn();
  const error = {
    message: `ERROR!`,
    imageUrl: `http://imgerror.url/err.gif`
  };
  const openModal = jest.fn();
  const results = [
    image1,
    image2,
    image3
  ]
  const component = renderer.create(
    <SearchResults
      copyUrl={copyUrl}
      openModal={openModal}
      results={results}
    />
  );

  it(`renders properly`, () => {
    expect(component).toMatchSnapshot();
  });

  it(`renders properly with error`, () => {
    const component = renderer.create(
      <SearchResults
        copyUrl={copyUrl}
        error={error}
        openModal={openModal}
        results={results}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
