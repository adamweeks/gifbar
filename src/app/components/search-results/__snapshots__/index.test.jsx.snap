// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults component renders properly 1`] = `
<div
  className="image-grid"
>
  <div
    className="col"
  >
    <div
      className="image-container"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "height": "50px",
        }
      }
    >
      <img
        onClick={[Function]}
        src="http://mycoolserver.info/kid-n-play.gif"
      />
    </div>
    <div
      className="image-container"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "height": "50px",
        }
      }
    >
      <img
        onClick={[Function]}
        src="http://mycoolserver.info/kid-n-play.gif"
      />
    </div>
  </div>
  <div
    className="col"
  >
    <div
      className="image-container"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "height": "50px",
        }
      }
    >
      <img
        onClick={[Function]}
        src="http://mycoolserver.info/kid-n-play.gif"
      />
    </div>
  </div>
</div>
`;

exports[`SearchResults component renders properly with error 1`] = `
<div>
  <p>
    ERROR!
  </p>
  <p>
    <img
      src="http://imgerror.url/err.gif"
    />
  </p>
</div>
`;
